docs/concepts/callbacks.md

```diff
@@ -67,3 +69,67 @@ A callback is sent to https://example.com/custom with the following JSON body:
 }
 """
 ```
+
+## Authentication for callbacks
+
+!!! Warning
+    This feature is currently in beta, and the API is likely to change.
+
+Callbacks can be authenticated using shared authentication headers. To enable authentication,
+set either `default_callback_auth_kind` when creating the endpoint or `callback_auth_kind`
+when making a prediction request.
+
+Currently, the supported authentication methods are `basic` and `mtls`. If `basic` is used,
+then the `default_callback_auth_username` and `default_callback_auth_password` fields must be
+specified when creating the endpoint, or the `callback_auth_username` and `callback_auth_password`
+fields must be specified when making a prediction request. If `mtls` is used, then the 
+same is true for the `default_callback_auth_cert` and `default_callback_auth_key` fields,
+or the `callback_auth_cert` and `callback_auth_key` fields.
+
+```py title="Creating an Async Model Endpoint with custom Callback auth"  hl_lines="18-21 37-39"
+import os
+import time
+from launch import EndpointRequest, LaunchClient, PostInferenceHooks
+
+client = LaunchClient(api_key=os.getenv("LAUNCH_API_KEY"))
+endpoint = client.create_model_endpoint(
+    endpoint_name="demo-endpoint-callback",
+    model_bundle="test-bundle",
+    cpus=1,
+    min_workers=1,
+    endpoint_type="async",
+    update_if_exists=True,
+    labels={
+        "team": "MY_TEAM",
+        "product": "MY_PRODUCT",
+    },
+    post_inference_hooks=[PostInferenceHooks.CALLBACK],
+    default_callback_url="https://example.com",
+    default_callback_auth_kind="basic",
+    default_callback_auth_username="user",
+    default_callback_auth_password="password",
+)
+
+while endpoint.status() != "READY":
+    time.sleep(10)
+
+future_default = endpoint.predict(
+    request=EndpointRequest(args={"x": 2, "y": "hello"})
+)
+"""
+A callback is sent to https://example.com with ("user", "password") as the basic auth.
+"""
+
```

Comment:
qq do we want an example of using mtls with a default cert/key? (for completeness sake)

Reply:
I guess the hope was that it would be clear from the examples in the other direction, but it probably wouldn't hurt.

Reply:
I'll just do it for completeness then